### PR TITLE
feat(libs-shared): [SKP-222] left side menu

### PR DESCRIPTION
**Pull request by SimasKir**

### CODE REVIEW

File: libs/shared/left-side-menu/src/lib/shared-left-side-menu.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+  const list = () => (
+    <div
+      role="presentation"
+      onClick={toggleDrawer(false)}
+      onKeyDown={toggleDrawer(false)}
+    >
+      <List>
+        {menuItems.map((item, index) => (
+          <ListItem button key={index} href={item.path} className="list-item">
```
Comment (Gardemarin) [inline libs/shared/left-side-menu/src/lib/shared-left-side-menu.tsx]: You use `href` so need to add `component="a"` to this line

File: libs/shared/left-side-menu/src/lib/shared-left-side-menu.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  name: string;
+  path?: string
+}
+
+export const LeftSideMenu = (props: LeftSideMenuProps) => {
+  const [state, setState] = React.useState({
+    menuOpen: true,
+  });
+
+  const menuItems = props.menuItemsArray;
```
Comment (Gardemarin) [inline libs/shared/left-side-menu/src/lib/shared-left-side-menu.tsx]: Use   
```
useEffect(() => {

  }, [props]); 
```
to load menuItems

File: libs/shared/left-side-menu/src/lib/shared-left-side-menu.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        'path': '/'
+      },
+      {
+        'name': 'Blockchain',
+        'path': '/'
+      }
+    ]
+  );
+
+  /* eslint-disable-next-line */
```
Comment (Gardemarin) [inline libs/shared/left-side-menu/src/lib/shared-left-side-menu.stories.tsx:49]: Delete this comment

File: libs/shared/left-side-menu/src/lib/shared-left-side-menu.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+export const SharedLeftSideMenu = (props: SharedLeftSideMenuProps) => {
+         const [state, setState] = React.useState({
+           menuOpen: true,
+         });
+
+         const menuItems = props.menuItemsArray;
+
+         const toggleDrawer = (open: boolean) => (
+           event: React.KeyboardEvent | React.MouseEvent
```
Comment (doncevas) [inline libs/shared/left-side-menu/src/lib/shared-left-side-menu.tsx:30]: don't use close drawer on menu click, i recommend use extra button on top for opening and closing drawer. ex: https://material-ui.com/components/drawers/#persistent-drawer

File: libs/shared/left-side-menu/src/lib/shared-left-side-menu.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  };
+
+  const list = () => (
+    <div
+      role="presentation"
+      onClick={toggleDrawer(false)}
+      onKeyDown={toggleDrawer(false)}
+    >
+      <List>
+        {menuItems.map((item, index) => (
```
Comment (Gardemarin) [inline libs/shared/left-side-menu/src/lib/shared-left-side-menu.tsx]: check is `menuItems` not null, `!!menuItems`

File: libs/shared/left-side-menu/src/lib/shared-left-side-menu.tsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+  })
+
+  const [menuItems, setMenuItems] = useState([]);
+
+  useEffect(() => {
+    setMenuItems(props.menuItemsArray);
+  }, [props]);
+
+  const toggleDrawer = (open: boolean) => (
+    event: React.KeyboardEvent | React.MouseEvent
```
Comment (doncevas) [inline libs/shared/left-side-menu/src/lib/shared-left-side-menu.tsx:41]: this event useless, remove it

File: libs/shared/left-side-menu/src/lib/shared-left-side-menu.tsx [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    </div>
+  );
+
+  return (
+    <div>
+      <IconButton
+        color="inherit"
+        aria-label="open drawer"
+        onClick={toggleDrawer(true)}
+        // edge="start"
```
Comment (Gardemarin) [inline libs/shared/left-side-menu/src/lib/shared-left-side-menu.tsx:76]: Remove this comment if you don't use it, but if you will use it, change comment to `// TODO: edge="start"`

File: libs/shared/left-side-menu/src/lib/shared-left-side-menu.tsx [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+  path: string
+}
+
+export interface SharedLeftSideMenuProps {
+  title: string,
+  menuItemsArray: Array<Type>
+}
+
+export const SharedLeftSideMenu = (props: SharedLeftSideMenuProps) => {
+  const [state, setState] = useState({
```
Comment (doncevas) [inline libs/shared/left-side-menu/src/lib/shared-left-side-menu.tsx:30]: rename state to isMenuOpen, setMenuOpen and on state set value not object useState(false)

File: libs/shared/left-side-menu/src/lib/shared-left-side-menu.tsx [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+    menuOpen: false,
+  })
+
+  const [menuItems, setMenuItems] = useState([]);
```
Comment (doncevas) [inline libs/shared/left-side-menu/src/lib/shared-left-side-menu.tsx:34]: what for this state? you can map menu directly from props

File: libs/shared/left-side-menu/src/lib/shared-left-side-menu.tsx (hunk 2 above)
Comment (doncevas) [reply libs/shared/left-side-menu/src/lib/shared-left-side-menu.tsx:32]: you can directly use props.menuItemsArray on map, no need reassign to variable.

**Pull request merged**